docs/technical.rst

```diff
@@ -95,8 +95,26 @@ Features supported, encouraged and discouraged from the different ``JavaScript``
 TypeScript
 ^^^^^^^^^^
 
-``TODO``
+TypeScript is a type-safe version of JavaScript that compiles down to JavaScript. Type-safety
+offers a better development experience and improves the reliability and maintainability of
+large projects. TypeScript also produces type definition files to allow upstream projects to
+safely use EthereumJS libraries.
+
+TypeScript language support is a compile-time feature separate from the runtime JavaScript support.
+All EthereumJS projects that are written in TypeScript are compiled using TypeScript 3.2 or newer.
+
+Notable language features:
+
+- Async Iteration
+
+- `Generics <https://www.typescriptlang.org/docs/handbook/generics.html>`_
+
+- `Function Overload <https://www.typescriptlang.org/docs/handbook/functions.html>`_
+
+Discouraged language features:
 
+The use of the ``any`` type when writing TypeScript in an EthereumJS project is discourgaed.
+The use of experimental features are also discouraged as they may change in future releases.
```

Comment:
Maybe also do this as a list here. This encourages follow-up documentation editors to just add or remove bullet points, in this form one has to always somewhat rewrite the whole text (which is no fun 😄).

Also better to keep it short (just "Use of ``any`` type", everything else is redundant), the shorter and comprise a documentation is the higher the chance that someone reads it.

Reply:
> Maybe also do this as a list here. This encourages follow-up documentation editors to just add or remove bullet points, in this form one has to always somewhat rewrite the whole text [...].

Good point, as you noted above about the BigInt support, I'll list out some of the experimental features we're looking to use once they become stable.